lib/hamt.ml

```diff
@@ -212,7 +212,7 @@ module Make (Config : CONFIG) (Key : Hashtbl.HashedType) :
 
   let rec combine_tip shift node1 node2 =
     match (node1, node2) with
-    | Leaf (h1, k1, v1), Leaf (h2, k2, v2) when h1 = h2 ->
+    | Leaf (h1, k1, v1), Leaf (h2, k2, v2) when Int.equal h1 h2 ->
```

Comment:
For some reason Travis does not work, but this will break before 4.08 and the introduction of the `Int` module. If we want to keep compatibility this far, and since most our equalities are actually tested on integers, what about `let int_equal (x : int) y = x = y` ?

Reply:
Yes indeed. Will do.